### PR TITLE
Update UserGroupManager

### PR DESCRIPTION
**Pull request by thecrimpmaster**
Allow user group update by name. Allow update of user group roles

### CODE REVIEW

**Comment by marek-pietrzak-tg** [comment]
pretty massive method 🙈

File: Core/API/Managers/UserGroupManager.php [hunk 1]
```diff
@@ -69,16 +73,33 @@ public function create()
...
         $contentService = $this->repository->getContentService();
 
-        $userGroup = $userService->loadUserGroup($this->dsl['id']);
+        $userGroup = null;
+        
+        if (array_key_exists('id', $this->dsl)) {
+            $userGroup = $userService->loadUserGroup($this->dsl['id']);
+        }
+
+        if (array_key_exists('name', $this->dsl)) {
```
Comment (phalasz) [inline Core/API/Managers/UserGroupManager.php]: shouldn't this be an `else if` instead?

Comment (phalasz) [inline Core/API/Managers/UserGroupManager.php]: or check if we already have assigned something to `$userGroup`

**Comment by marek-pietrzak-tg** [comment]
👍

**Pull request merged**